src/nodes.coffee

```diff
@@ -3498,6 +3506,35 @@ exports.Op = class Op extends Base
   toString: (idt) ->
     super idt, @constructor.name + ' ' + @operator
 
+  ast: ->
+    @checkUpdateAssignability()
+    super()
+
+  astType: ->
+    switch @operator
+      when 'new'           then 'NewExpression'
+      when '||', '&&', '?' then 'LogicalExpression'
```

Comment:
Had realized that binary `?` really behaved more like a `LogicalExpression` (ie `and`/`or`/`&&`/`||`) than a `BinaryExpression` in that it has similar control flow semantics. And then saw that the proposed JS version of this operator (`??`) does in fact generate a `LogicalExpression` eg in Babel